In [18]:
!pip install lazypredict
import pandas as pd
from sklearn.model_selection import train_test_split
from lazypredict.Supervised import LazyClassifier, LazyRegressor


In [19]:
# Load a dataset (example: Breast Cancer dataset)
from sklearn.datasets import load_breast_cancer
data = load_breast_cancer()

# Convert to DataFrame
X = pd.DataFrame(data.data, columns=data.feature_names)
y = pd.Series(data.target)

# Split into train and test sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

In [20]:
# Step 6: Import LazyPredict and necessary modules
from lazypredict.Supervised import LazyClassifier
from sklearn.model_selection import train_test_split
from sklearn.metrics import precision_score, recall_score


lazy_classifier = LazyClassifier(verbose=0, ignore_warnings=True, custom_metric=None)
models, predictions = lazy_classifier.fit(X_train, X_test, y_train, y_test)

# Step 10: Calculate Precision and Recall for each model manually
precision_scores = []
recall_scores = []

# Loop through each model in LazyPredict and calculate precision and recall
for model_name in models.index:
    model = lazy_classifier.models[model_name]  # Get the trained model
    y_pred = model.predict(X_test)  # Predict on the test set

    # Calculate precision and recall for the current model
    precision = precision_score(y_test, y_pred, average='binary', zero_division=0)
    recall = recall_score(y_test, y_pred, average='binary', zero_division=0)

    precision_scores.append(precision)
    recall_scores.append(recall)

# Step 11: Add Precision and Recall scores to the 'models' DataFrame
models['Precision'] = precision_scores
models['Recall'] = recall_scores

# Step 12: Display the updated comparison of models
print(models)

 97%|█████████▋| 30/31 [00:05<00:00,  2.24it/s]

[LightGBM] [Info] Number of positive: 286, number of negative: 169
[LightGBM] [Info] Auto-choosing col-wise multi-threading, the overhead of testing was 0.000305 seconds.
You can set `force_col_wise=true` to remove the overhead.
[LightGBM] [Info] Total Bins 4540
[LightGBM] [Info] Number of data points in the train set: 455, number of used features: 30
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.628571 -> initscore=0.526093
[LightGBM] [Info] Start training from score 0.526093
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best 

100%|██████████| 31/31 [00:06<00:00,  4.90it/s]


                               Accuracy  Balanced Accuracy  ROC AUC  F1 Score  \
Model                                                                           
BernoulliNB                        0.98               0.98     0.98      0.98   
PassiveAggressiveClassifier        0.98               0.98     0.98      0.98   
SVC                                0.98               0.98     0.98      0.98   
Perceptron                         0.97               0.97     0.97      0.97   
AdaBoostClassifier                 0.97               0.97     0.97      0.97   
LogisticRegression                 0.97               0.97     0.97      0.97   
SGDClassifier                      0.96               0.97     0.97      0.97   
ExtraTreeClassifier                0.96               0.97     0.97      0.97   
CalibratedClassifierCV             0.97               0.97     0.97      0.97   
RandomForestClassifier             0.96               0.96     0.96      0.96   
LGBMClassifier              

In [21]:
# Initialize LazyRegressor
reg = LazyRegressor(verbose=0, ignore_warnings=True, custom_metric=None)

# Run LazyPredict
models, predictions = reg.fit(X_train, X_test, y_train, y_test)

# Display Results
print(models)

 98%|█████████▊| 41/42 [00:15<00:01,  1.15s/it]

[LightGBM] [Info] Auto-choosing col-wise multi-threading, the overhead of testing was 0.000308 seconds.
You can set `force_col_wise=true` to remove the overhead.
[LightGBM] [Info] Total Bins 4540
[LightGBM] [Info] Number of data points in the train set: 455, number of used features: 30
[LightGBM] [Info] Start training from score 0.628571
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, b

100%|██████████| 42/42 [00:15<00:00,  2.67it/s]


[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No 

here i will go with svc model because it has accuracy of 98% and precession:0.97 and recall:1.00